runtime/gc_glue_java/ArrayletObjectModel.hpp

```diff
@@ -1222,17 +1227,23 @@ class GC_ArrayletObjectModel : public GC_ArrayletObjectModelBase
 	}
 
 	/**
-	 * Returns the address of first data slot in the array
-	 * @param arrayPtr Ptr to an array
+	 * Returns the address of first data slot in the array.
+	 *
+	 * @param arrayPtr Pointer to the contigous indexable object whose dataPointer we are trying to access
 	 * @return Address of first data slot in the array
 	 */
 	MMINLINE void *
 	getDataPointerForContiguous(J9IndexableObject *arrayPtr)
 	{
-		return (void *)((uintptr_t)arrayPtr + contiguousIndexableHeaderSize());
+		void *dataAddr = (void *)((uintptr_t)arrayPtr + contiguousIndexableHeaderSize());;
+#if defined(J9VM_GC_ENABLE_SPARSE_HEAP_ALLOCATION)
+		if (isVirtualLargeObjectHeapEnabled()) {
```

Comment:
it's probably more correct to do isDataAddressPresent check

Reply:
disregard the comment - isVirtualLargeObjectHeapEnabled is more specific/deterministic (and consistent with what DDR does)